tests/integration_tests/test_hyperopt.py

```diff
@@ -105,6 +119,60 @@ def _setup_ludwig_config(dataset_fp: str) -> Tuple[Dict, str]:
     return config, rel_path
 
 
+def _setup_ludwig_config_with_shared_params(dataset_fp: str) -> Tuple[Dict, str]:
+    input_features = [
+        text_feature(name="title", encoder="parallel_cnn"),
+        text_feature(name="summary"),
+        category_feature(vocab_size=3),
+        category_feature(vocab_size=3),
+    ]
+
+    output_features = [category_feature(vocab_size=3)]
+
+    rel_path = generate_data(input_features, output_features, dataset_fp)
+
+    num_filters_search_space = [4, 8]
+    embedding_size_search_space = [4, 8]
+
+    # Add default parameters in hyperopt parameter search space
+    config = {
+        INPUT_FEATURES: input_features,
+        OUTPUT_FEATURES: output_features,
+        COMBINER: {TYPE: "concat", "num_fc_layers": 2},
+        TRAINER: {"epochs": 2, "learning_rate": 0.001},
+        HYPEROPT: {
+            "parameters": {
+                "trainer.learning_rate": {"lower": 0.0001, "upper": 0.01, "space": "loguniform"},
+                "defaults.input_features.text.num_filters": {"space": "choice", "categories": num_filters_search_space},
+                "defaults.output_features.category.embedding_size": {
+                    "space": "choice",
+                    "categories": embedding_size_search_space,
+                },
+            },
+            "goal": "minimize",
+            "output_feature": output_features[0][NAME],
+            "validation_metrics": "loss",
+            "executor": {"type": "ray", "num_samples": RANDOM_SEARCH_SIZE},
+            "search_alg": {"type": "variant_generator"},
+        },
+    }
+
+    return config, rel_path, num_filters_search_space, embedding_size_search_space
+
+
+def _get_trial_parameter_value(parameter_key: str, trial_row: str) -> Union[str, None]:
+    """Returns the parameter value from the Ray trial row, which has slightly different column names depending on
+    the version of Ray. Returns None if the parameter key is not found.
+
+    TODO(#2176): There are different key name delimiters depending on Ray version. Simplify this as Ray is upgraded.
+    The delimiter in future versions of Ray will be '/' instead of '.'
+    """
+    if f"config.{parameter_key}" in trial_row:
```

Comment:
Instead of checking the format of `config` + `parameter key`, can you make a check based on the ray version? [Here's an example](https://github.com/ludwig-ai/ludwig/blob/a5871819bc430b79ec97ed12724dab157874fb78/tests/integration_tests/test_hyperopt_ray_horovod.py#L61). Also specifying which ray version we need to update in the comment would be helpful.

Reply:
Good suggestion, will update!